addons/binding/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/handler/PentairSerialBridgeHandler.java

```diff
@@ -0,0 +1,215 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gnu.io.CommPort;
+import gnu.io.CommPortIdentifier;
+import gnu.io.NoSuchPortException;
+import gnu.io.PortInUseException;
+import gnu.io.SerialPort;
+import gnu.io.UnsupportedCommOperationException;
+
+/**
+ * Handler for the IPBridge. Implements the connect and disconnect abstract methods of {@link PentairBaseBridgeHandler}
+ *
+ * @author Jeff James - initial contribution
+ *
+ */
+public class PentairSerialBridgeHandler extends PentairBaseBridgeHandler {
+    private Logger logger = LoggerFactory.getLogger(PentairSerialBridgeHandler.class);
+
+    /** SerialPort object representing the port where the RS485 adapter is connected */
+    SerialPort port;
+
+    public PentairSerialBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    protected synchronized void connect() {
+        PentairSerialBridgeConfig configuration = getConfigAs(PentairSerialBridgeConfig.class);
+
+        try {
+            updateSerialProperties(configuration.serialPort);
+            CommPortIdentifier ci = CommPortIdentifier.getPortIdentifier(configuration.serialPort);
+            CommPort cp = ci.open("openhabpentairbridge", 10000);
+            if (cp == null) {
+                throw new IllegalStateException("cannot open serial port!");
+            }
+
+            if (cp instanceof SerialPort) {
+                port = (SerialPort) cp;
+            } else {
+                throw new IllegalStateException("unknown port type");
+            }
+            port.setSerialPortParams(9600, SerialPort.DATABITS_8, SerialPort.STOPBITS_1, SerialPort.PARITY_NONE);
+            port.disableReceiveFraming();
+            port.disableReceiveThreshold();
+
+            reader = new BufferedInputStream(port.getInputStream());
+            writer = new BufferedOutputStream(port.getOutputStream());
+            logger.info("Pentair Bridge connected to serial port: {}", configuration.serialPort);
+        } catch (PortInUseException e) {
+            String msg = String.format("cannot open serial port: %s", configuration.serialPort);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
+            return;
+        } catch (UnsupportedCommOperationException e) {
+            String msg = String.format("got unsupported operation %s on port %s", e.getMessage(),
+                    configuration.serialPort);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
+            return;
+        } catch (NoSuchPortException e) {
+            String msg = String.format("got no such port for %s", configuration.serialPort);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
+            return;
+        } catch (IllegalStateException e) {
+            String msg = String.format("got unknown port type for %s", configuration.serialPort);
```

Comment:
This message does not seem completely in line with the exception and all exceptions contain a message which you ignore now, but could offer valuable information